File: day-03/part-1/th-ch.py
from tool.runners.python import SubmissionPy


class ThChSubmission(SubmissionPy):
    def run(self, input):
        nb_trees = 0
        x = 0
        y = 0
        lines = input.split("\n")
        for line in lines[:-1]:
            x += 3
            y += 1
            if lines[y][x % len(lines[y])] == "#":
                nb_trees += 1
        return nb_trees

File: day-13/part-1/th-ch.go
package main

import (
	"fmt"
	"io/ioutil"
	"os"
	"strconv"
	"strings"
	"time"
)

func run(s string) interface{} {
	lines := strings.Split(s, "\n")
	earliest, _ := strconv.Atoi(lines[0])

	bestBusID := 0
	bestWaitTime := earliest

	for _, id := range strings.Split(lines[1], ",") {
		if id == "x" {
			continue
		}

		bus, _ := strconv.Atoi(id)
		earliestDeparture := -earliest + (int(earliest/bus)+1)*bus
		if earliestDeparture < bestWaitTime {
			bestWaitTime = earliestDeparture
			bestBusID = bus
		}
	}

	return bestWaitTime * bestBusID
}

func main() {
	// Uncomment this line to disable garbage collection
	// debug.SetGCPercent(-1)

	// Read input from stdin
	input, err := ioutil.ReadAll(os.Stdin)
	if err != nil {
		panic(err)
	}

	// Start resolution
	start := time.Now()
	result := run(string(input))

	// Print result
	fmt.Printf("_duration:%f\n", time.Now().Sub(start).Seconds()*1000)
	fmt.Println(result)
}
